src/platform/plugins/shared/share/public/url_service/short_urls/short_url_client.ts

```diff
@@ -75,9 +76,30 @@ export class BrowserShortUrlClient implements IShortUrlClient {
   }
 
   public async createWithLocator<P extends SerializableRecord>(
-    params: ShortUrlCreateParams<P>
+    params: ShortUrlCreateParams<P>,
+    isAbsoluteTime?: boolean
   ): Promise<ShortUrlCreateResponse<P>> {
-    const result = await this.create(params);
+    const getUpdatedParams = (inputParams: ShortUrlCreateParams<P>) => {
+      const timeRange = inputParams.params?.timeRange as SerializableRecord;
+      if (isAbsoluteTime)
+        return {
+          ...inputParams,
+          params: {
+            ...inputParams.params,
+            timeRange: {
```

Comment:
I'm a little concerned that were injecting a `timeRange` on the `LocatorParams` even if the `timeRange` const on line 83 could be undefined such as in [DashboardLocatorParams](https://github.com/elastic/kibana/blob/0a60ccce642f787b97901c45e3d4848bb2381ae8/src/platform/plugins/shared/dashboard/common/types.ts#L82). Would it be possible to make the conditional something like `if (isAbsoluteTime && timeRange) {`